### PR TITLE
Extend statsd integration to configure relay endpoint

### PR DESCRIPTION
**Pull request by arminaaki**
#### PR Description

`statsd_exporter` natively supports setting the relay endpoint. However, the current implementation of statsd integration doesn't allow configuring the endpoint.

The proposed PR extends statsd integration to configure relay endpoint.

#### Notes to the Reviewer

The default value for `RelayPacketLength` was taken from the original implementation. Please refer to the code reference [here](https://github.com/prometheus/statsd_exporter/blob/c3752cf30f14d53838a3104c1fd1b20f3cc354eb/main.go#L262).

#### PR Checklist

- [x] CHANGELOG updated
- [X] Documentation added
- [X] Tests updated

### CODE REVIEW

**Review by clayton-cornell — COMMENTED** [review]
This looks OK from a docs standpoint.

**Review by jcreixell — APPROVED** [review]
LGTM, @arminaaki feel free to add a [CHANGELOG](https://github.com/grafana/agent/blob/main/CHANGELOG.md?plain=1#L12) entry to under "Main (unreleased)", "Enhancements" subsection with your github handle.

I can merge it afterwards

File: docs/sources/flow/reference/components/prometheus.exporter.statsd.md [hunk 1]
```diff
@@ -42,6 +42,8 @@ Name | Type | Description | Default | Required
 `parse_influxdb_tags`                             | `string`       | Parse InfluxDB style tags. | `true`| no
 `parse_librato_tags`                              | `string`       | Parse Librato style tags. | `true`| no
 `parse_signalfx_tags`                             | `string`       | Parse SignalFX style tags. | `true`| no
+`relay_addr`                                      | `string`       | Relay address configuration(UDP endpoint in the format 'host:port'). | | no
```
Comment (clayton-cornell) [inline docs/sources/flow/reference/components/prometheus.exporter.statsd.md:45]:
```suggestion
`relay_addr`                                      | `string`       | Relay address configuration (UDP endpoint in the format 'host:port'). | | no
```

**Comment by arminaaki** [comment]
Thank you for the review. Changelog is now updated!

**Review by jcreixell — APPROVED** [review]

**Pull request merged**